oneclick.py

```diff
@@ -367,6 +369,96 @@ def get_metadata_by_isbn(self, identifier):
         return respdict
 
 
+    def populate_all_catalog(self):
+        """ Call get_all_catalog to get all of library's book info from OneClick.
+        Create Work, Edition, LicensePool objects in our database.
+        """
+        catalog_list = self.get_all_catalog()
+        items_transmitted = len(catalog_list)
+        items_created = 0
+        for catalog_item in catalog_list:
+            result = self.update_metadata(catalog_item)
+            if result:
+                items_created += 1
+
+        return items_transmitted, items_created
+
+
+    def populate_delta(self, months=1):
+        """ Call get_delta for the last month to get all of the library's book info changes 
+        from OneClick.  Update Work, Edition, LicensePool objects in our database.
+        """
+        today = datetime.datetime.now()
+        time_ago = relativedelta(months=months)
+
+        delta = self.get_delta(from_date=(today - time_ago), to_date=today)
+        if not delta or len(delta) < 1:
+            return None, None
+
+        items_added = delta[0].get("addedTitles", None)
+        items_removed = delta[0].get("removedTitles", None)
+
+        items_transmitted = len(items_added) + len(items_removed)
+        items_updated = 0
+        for catalog_item in items_added:
+            result = self.update_metadata(catalog_item)
+            if result:
+                items_updated += 1
+
+        for catalog_item in items_removed:
+            metadata = OneClickRepresentationExtractor.isbn_info_to_metadata(catalog_item)
+
+            if not metadata:
+                # generate a CoverageFailure to let the system know to revisit this book
+                # TODO:  if did not create a Work, but have a CoverageFailure for the isbn, 
+                # check that re-processing that coverage would generate the work.
+                e = "Could not extract metadata from OneClick data: %r" % catalog_item
+                make_note = CoverageFailure(identifier, e, data_source=self.output_source, transient=True)
+
+            # convert IdentifierData into Identifier, if can
+            identifier, made_new = metadata.primary_identifier.load(_db=self._db)
+            if identifier and not made_new:
+                [work] = Work.from_identifiers(self._db, [identifier])
+                if work:
+                    for cr in work.coverage_records:
+                        self._db.delete(cr)
+                    for pool in work.license_pools:
+                        if pool.presentation_edition:
+                            for contribution in pool.presentation_edition.contributions:
+                                self._db.delete(contribution)
+                            self._db.delete(pool.presentation_edition.contributions[0])
```

Comment:
Once you delete all the presentation edition's contributions, why is there still one left to be deleted?
Why is it necessary to delete the presentation edition's contributions, but not the presentation edition itself?
I don't think you should be deleting the presentation edition at all--it might be used by another license pool for the same book.